interface/forms/newpatient/common.php

```diff
@@ -76,444 +76,459 @@ function sensitivity_compare($a, $b)
-            <?php
-            billing_facility('billing_facility', $result['billing_facility']);
-            ?>
-            </div>
-        </td>
-     </tr>
-        <?php if ($GLOBALS['set_pos_code_encounter']) { ?>
-        <tr>
-            <td><span class='bold' nowrap><?php echo xlt('POS Code'); ?>: </span></td>
-            <td colspan="6">
-                <select class="form-control" name="pos_code">
-                <?php
-
-                $pc = new POSRef();
-
-                foreach ($pc->get_pos_ref() as $pos) {
-                    echo "<option value=\"" . attr($pos["code"]) . "\" ";
-                    if ($pos["code"] == $result['pos_code'] || $pos["code"] == $posCode) {
-                        echo "selected";
-                    }
-
-                    echo ">" . text($pos['code'])  . ": ". xlt($pos['title']);
-                    echo "</option>\n";
-                }
-
-                ?>
-                </select>
-            </td>
-       </tr>
-        <?php } ?>
-    <tr>
-<?php
- $sensitivities = acl_get_sensitivities();
-if ($sensitivities && count($sensitivities)) {
-    usort($sensitivities, "sensitivity_compare");
-?>
-   <td class='bold' nowrap><?php echo xlt('Sensitivity:'); ?></td>
-    <td class='text'>
-     <select class="form-control" name='form_sensitivity'>
-<?php
-foreach ($sensitivities as $value) {
-   // Omit sensitivities to which this user does not have access.
-    if (acl_check('sensitivities', $value[1])) {
-        echo "       <option value='" . attr($value[1]) . "'";
-        if ($viewmode && $result['sensitivity'] == $value[1]) {
-            echo " selected";
+        var target = window;
+        while (target != top) {
+            if (target.name == 'RBot') {
+                target.parent.left_nav.loadFrame('ens1', window.name, 'patient_file/history/encounters.php');
+                break;
+            }
+            else if (target.name == 'RTop') {
+                target.parent.left_nav.loadFrame('dem1', window.name, 'patient_file/summary/demographics.php');
+                break;
+            }
+            target = target.parent;
         }
-
-        echo ">" . xlt($value[3]) . "</option>\n";
+        return false;
     }
-}
 
-echo "       <option value=''";
-if ($viewmode && !$result['sensitivity']) {
-    echo " selected";
+    // Handler for cancel clicked when not creating a new encounter.
+    // Just reload the view mode.
+    function cancelClickedOld() {
+        location.href = '<?php echo "$rootdir/patient_file/encounter/forms.php"; ?>';
+        return false;
+    }
+    
+</script>
+<style>
+@media only screen and (max-width: 1024px) {
+    #visit-details [class*="col-"], #visit-issues [class*="col-"]{
+    width: 100%;
+    text-align:left!Important;
 }
-
-echo ">" . xlt('None'). "</option>\n";
-?>
-     </select>
-    </td>
+</style>
 <?php
+if ($viewmode) {
+    $body_javascript = '';
+    $heading_caption = 'Patient Encounter Form';
```

Comment:
xl() it here (to ensure it goes to translation pipeline)
(and then html escape it later)

Reply:
will fix

Reply:
Just for my info, how close to the html element does it need to be escaped and translated ?

Reply:
2 issue here:
1. If possible, always best to translate the actual string rather than a variable (the perl script that collects constants can't pick them up if a variable); of course, there are times when this is not possible (sometimes I'll then put them in a comment).
2. for html escaping as downstream as possible (goal is to do them only when output them to html); this avoids using escaping variable sin algorithms or sending them to databases or non html output.